typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -87,22 +134,37 @@ export class EvmERC20WarpRouteReader {
       routerAddress,
       this.provider,
     );
-    const [mailbox, owner, token, hook, interchainSecurityModule] =
-      await Promise.all([
-        warpRoute.mailbox(),
-        warpRoute.owner(),
-        warpRoute.wrappedToken(),
-        warpRoute.hook(),
-        warpRoute.interchainSecurityModule(),
-      ]);
+    const [mailbox, owner, hook, interchainSecurityModule] = await Promise.all([
+      warpRoute.mailbox(),
+      warpRoute.owner(),
+      warpRoute.hook(),
+      warpRoute.interchainSecurityModule(),
+    ]);
 
-    return {
+    const metadata: WarpRouteBaseMetadata = {
       mailbox,
       owner,
-      token,
       hook,
-      interchainSecurityModule,
     };
+
+    isZeroishAddress(interchainSecurityModule)
```

Comment:
noob question but what does a zero address ISM mean?

Reply:
good question :)

since we are fetching `interchainSecurityModule` from the warp route contract (in the `promise.all` above), if there are any addresses that are not "set", then solidity by default will return the zero address. in other words, the ISM was not set on the Warp Route contract

Reply:
can you leave add that as a comment in code pls, to be able to remember this in the future

Reply:
ethersjs isnt returning zero address by "default", the address is actually zero in the storage slot onchain
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/0cf692e7315cb91059a4c0b39c7cdf76151851cc/solidity/contracts/Mailbox.sol#L408 we switch on this in the Mailbox to check whether to use the default ISM or a recipient override

Reply:
and I would interpret zeroish address as "undefined" on the ism config

Reply:
Fixed.